staff/diana-moreno/precourse/airlines/airlines-pro/index-oop.js

```diff
@@ -0,0 +1,294 @@
+      if(arr[i][prop] === num) {
+        return i;
+      }
+    }
+  }
+
+  userSelectPrice() {
+    let search = prompt("Introduzca un precio para buscar y comparar vuelos")
+    while(this.isString(search)) {
+      alert("El precio del vuelo debe contener unicamente valores numericos")
+      search = prompt("Introduzca un precio para buscar vuelos")
+    };
+    return search;
+  }
+
+  searchFlightsByPrice(arr, prop) {
+    let search = this.userSelectPrice()
+    for(let i in arr) {
+      if(arr[i][prop] > Number.parseFloat(search)) {
+        this.moreExpensiveFlights.push(arr[i])
+      } else if(arr[i][prop] < Number.parseFloat(search)) {
+        this.lessExpensiveFlights.push(arr[i])
+      } else {
+        this.equalExpensiveFlights.push(arr[i])
+      }
+    }
+    this.lessExpensiveFlights.sort((a, b) => a[prop] - b[prop])
+    this.moreExpensiveFlights.sort((a, b) => a[prop] - b[prop])
+  }
+
+  showFlightsByPrice(arr, string) {
+    arr.length <= 0 ? console.log(`No hay vuelos con precio ${string} al seleccionado`) : console.log(`\nLos vuelos con precio ${string} al seleccionado son: `);
+    for(let i in arr) {
+      let scale = arr[i].scale ? `no realiza ninguna escala` : `realiza escala`
+      console.log(`Id: ${arr[i].id} | Vuelo con origen ${arr[i].from} y destino ${arr[i].to}, precio de ${arr[i].cost} euros y ${scale} `);
+     }
+  }
+
+  buyFlight(arr) {
+    let buy = confirm("Quiere comprar un vuelo?");
+    if(buy) {
+      let flightToBuy = prompt("Que vuelo quiere comprar?\n\nId de vuelos disponibles: " + this.idDisponible(arr))//cambio flights por arr
+      let flightBought = this.searchIndex(arr, "id", Number(flightToBuy))
+      let idDispo = this.idDisponible(arr);
+      if(idDispo.includes(Number(flightToBuy))) {
+        let finalConfirmation = confirm(`El vuelo seleccionado es:\n\nId: ${arr[flightBought].id} | Vuelo con origen ${arr[flightBought].from} y destino ${arr[flightBought].to}, precio de ${arr[flightBought].cost} euros\n\nEs correcto?`);
+        finalConfirmation ? alert("Gracias por su compra, vuelva pronto")
+                        : this.buyFlight(arr);
+      } else {
+        alert("El id introducido no es correcto")
+        this.buyFlight(arr);
+      }
+    } else {
+      alert("Ok, bye!");
+    }
+  }
+}
+
+class Airlines {
+  constructor() {
+    this.greeting();
+    this.arrayFlights = new Flights();
+    let flights = this.arrayFlights.flights
+    this.infoFlights = new InfoFlights(flights);
+    this.credentials = new Credentials();
+    if(this.credentials.typeOfUser === "admin") {
+      this.isAdmin = new Admin(flights);
+    } else if(this.credentials.typeOfUser === "user") {
+      this.isUser = new User(flights)
+    }
+  }
+
+  greeting() {
+    let name = prompt("Bienvenido a Skylab Airlines, cual es su nombre?") || "anonimo";
+    alert("Hola " + name + " !\n\nA continuacion se mostraran los vuelos disponibles para hoy.");
+  }
+}
+
+new Airlines();
```

Comment:
Very clear, with an readable and interesting code since it is focused on OOP, I need to check OOP and your code will help me to undestand it!